File: PrimeFactors.dart


List<int> divisors(int integer) {
  List<int> factors = [];
  for (int i = 2; i <= integer ~/ 2; i++) {
    if (integer % i == 0) {
      factors.add(i);
      if (i != integer ~/ i) {
        factors.add(integer ~/ i);
      }
    }
  }
  if (factors.isEmpty) {
    return [integer]; 
  }
  return factors;
}

void main() {
  print(divisors(12)); // Output: [2, 3, 4, 6]
  print(divisors(25)); // Output: [5]
  print(divisors(13)); // Output: [13] (since 13 is prime)
}

File: PrimeFactors.py


def divisors(integer):
    factors = []
    for i in range(2,integer):
        if integer % i == 0:
            factors.append(i)
    if factors == []:
        return f'{integer} is prime'
    return factors

print(divisors(12)) #should return [2,3,4,6]
print(divisors(25)) #should return [5]
print(divisors(13)) #should return "13 is prime"
